core/electron/src/test/backend/ElectronHost.test.ts

```diff
@@ -157,49 +153,56 @@ async function testMainWindowUrl() {
   assert(url === window.webContents.getURL());
 }
 
-async function _testWindowSizeSettings() {
+async function testWindowSizeSettings() {
   const storeWindowName = "settingsTestWindow";
 
   await ElectronHost.startup();
+
+  NativeHost.settingsStore.removeData(`windowMaximized-${storeWindowName}`);
+  NativeHost.settingsStore.removeData(`windowPos-${storeWindowName}`);
+
   await ElectronHost.openMainWindow({ storeWindowName });
 
   const window = ElectronHost.mainWindow;
   assert(window);
 
   let size = ElectronHost.getWindowSizeSetting(storeWindowName);
-  const expectedSize = ElectronHost.getWindowSizeSetting(storeWindowName);
-  assert(size?.width === expectedSize?.width);
-  assert(size?.height === expectedSize?.height);
-  assert(size?.x === expectedSize?.x);
-  assert(size?.y === expectedSize?.y);
+  const expectedSize = window.getSize();
+  const expectedPos = window.getPosition();
+  assert(size?.width === expectedSize?.[0]);
+  assert(size?.height === expectedSize?.[1]);
+  assert(size?.x === expectedPos?.[0]);
+  assert(size?.y === expectedPos?.[1]);
 
   let isMaximized = ElectronHost.getWindowMaximizedSetting(storeWindowName);
   assert(isMaximized === window.isMaximized());
 
   window.maximize();
-  await new Promise((resolve) => setTimeout(resolve, 100)); // "maximize" event is not always emitted immediately
+  window.emit("maximize"); // "maximize" event is not emitted when running with xvfb
```

Comment:
It does. 

We could check if we are using `xvfb` before emitting event but that is not completely straight forward. It is easy enough to check if it is running on `linux` (I will add this check) but it is harder to check if `xvfb` is being used. The only way I can think of is using `pgreb xvfb` and all it will show us is that `xvfb` is running somewhere (although it should be enough).

Reply:
Makes sense - that would be good to add if not too much trouble.

Reply:
Added `isXvfbProcessRunning` for checking if `xvfb` is running.